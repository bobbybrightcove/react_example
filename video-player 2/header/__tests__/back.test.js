import React from 'react';
import renderer from 'react-test-renderer';
import BackIcon from '../back';

describe('BackIcon', () => {
  it('renders according to valid snapshot', () => {
    const component = renderer.create(
      <BackIcon screenId="better-call-saul" />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should change icon on mouse enter/leave', () => {
    const component = renderer.create(
      <BackIcon screenId="better-call-saul" />
    );

    // onMouseEnter
    component.toJSON().props.onMouseEnter();

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // onMouseLeave
    tree.props.onMouseLeave();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
