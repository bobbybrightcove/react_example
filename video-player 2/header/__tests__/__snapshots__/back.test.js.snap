// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackIcon renders according to valid snapshot 1`] = `
<a
  className="iconBox"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Back"
    src="back.svg"
  />
</a>
`;

exports[`BackIcon should change icon on mouse enter/leave 1`] = `
<a
  className="iconBox"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Back"
    src="back-hover.svg"
  />
</a>
`;

exports[`BackIcon should change icon on mouse enter/leave 2`] = `
<a
  className="iconBox"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Back"
    src="back.svg"
  />
</a>
`;
